pkg/bpf/bpf_linux.go

```diff
@@ -662,3 +662,52 @@ func TestDummyProg(progType ProgType, attachType uint32) error {
 
 	return errno
 }
+
+type BpfMapInfo struct {
+	Type       uint32
+	Id         uint32
+	KeySize    uint32
+	ValueSize  uint32
+	MaxEntries uint32
+	MapFlags   uint32
+}
+
+// GetMapInfoByFd returns map info for a map which is pointed by the given fd.
+func GetMapInfoByFd(fd uint32) (*BpfMapInfo, error) {
+	info := BpfMapInfo{}
+	uba := struct {
+		bpfFd   uint32
+		infoLen uint32
+		info    uint64
+	}{
+		uint32(fd),
+		uint32(unsafe.Sizeof(info)),
+		uint64(uintptr(unsafe.Pointer(&info))),
+	}
+
+	var duration *spanstat.SpanStat
+	if option.Config.MetricsConfig.BPFSyscallDurationEnabled {
+		duration = spanstat.Start()
+	}
+
+	ret, _, err := unix.Syscall(
+		unix.SYS_BPF,
+		BPF_OBJ_GET_INFO_BY_FD,
+		uintptr(unsafe.Pointer(&uba)),
+		unsafe.Sizeof(uba),
+	)
+	runtime.KeepAlive(&uba)
+
+	if option.Config.MetricsConfig.BPFSyscallDurationEnabled {
+		metrics.BPFSyscallDuration.WithLabelValues(metricOpGetMapInfoByFD, metrics.Errno2Outcome(err)).Observe(duration.End(err == 0).Total().Seconds())
+	}
+
+	if err != 0 {
+		return nil, fmt.Errorf("Unable to get BPF map info by fd %d: %s", fd, err)
+	}
+	if ret != 0 {
+		return nil, fmt.Errorf("Unable to get BPF map info by fd %d: %d", fd, ret)
```

Comment:
nit: These look like they should be wrapping the original error to allow potential for introspecting / comparing the base error further up the stack. I believe we settled on `fmt.Errorf(".... %w")` as the proper way to do this.

Reply:
I believe I can only can wrap the first `fmt.Errorf()`, as the second is of int type.